src/app/services/user-search.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable, OnDestroy, Inject  } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { DataStore } from './data.store';
+import { AUTH_API_URL } from 'ngx-login-client';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class UserSearchService implements OnDestroy {
+
+  private apiResult; // stores the http observable returned from the api
+  private resultSubscription: Subscription; // stores the subscription to apiResult
```

Comment:
it should be named just `subscriptions` and its should be an array of `Subscription` so that other functions can also add their subscription to it and all of that can be destroyed in `ngOnDestroy`.